### PR TITLE
Added state change diagram

### PR DESCRIPTION
**Pull request by mthatt**
This state change diagram should help users visualize how states transition in Prefect. mermaid.js is ordering the states strangely with mintlify, particular the section with 'AwaitingRetry' and 'Retrying' on the left.

### Checklist

- [ ] This pull request references any related issue by including "closes `<link to issue>`"
  - If no issue exists and your change is not a small fix, please [create an issue](https://github.com/PrefectHQ/prefect/issues/new/choose) first.
- [ ] If this pull request adds new functionality, it includes unit tests that cover the changes
- [ ] If this pull request removes docs files, it includes redirect settings in `mint.json`.
- [ ] If this pull request adds functions or classes, it includes helpful docstrings.

### CODE REVIEW

**Comment by discdiver** [comment]
Thank you, @mthatt. The Mintlify preview deploy doesn't seem to be following, so not able to see a rendered version of the diagram. Are you able to add a screenshot from local?

**Comment by mthatt** [comment]
Screenshot 2024-11-01 at 5 40 55 PM

Here's a screenshot of how it looks in docs

**Comment by discdiver** [comment]
Cool! Re: the suboptimal retry placement. I'm not sure, but maybe it can be managed explicitly with divs?

**Review by discdiver — APPROVED** [review]

File: docs/3.0/develop/manage-states.mdx [hunk 1]
```diff
@@ -56,6 +56,109 @@ The final state of a flow or task run depends on a number of factors; generally
...
+    RolledBack[Rolled Back]:::completed
+    Crashed[Crashed]:::crashed
+
+    Paused[Paused]:::paused
+
+    Completed[Completed]:::completed
+
+    %% Connections
+    Scheduled --> |Scheduled start time passes without entering Pending| Late
+    Scheduled --> |Scheduled start time reached| Pending
```
Comment (cicdw) [inline docs/3.0/develop/manage-states.mdx:126]: I think this is a bit too cryptic to be useful - a Scheduled -> Pending transition means first and foremost that a worker successfully submitted the run.
```suggestion
    Scheduled --> |Worker/Runner successfully submits run| Pending
```
Open to other wording if that's not clear

**Review by cicdw — CHANGES_REQUESTED** [review]
one small request for a wording update, otherwise this is pretty helpful thank you

File: docs/3.0/develop/manage-states.mdx (hunk 1 above)
Comment (discdiver) [reply docs/3.0/develop/manage-states.mdx:126]: I chatted with @mthatt and he was on board with that change, so committing.

**Comment by discdiver** [comment]
> one small request for a wording update, otherwise this is pretty helpful thank you

I accepted your change @cicdw, if you want to approve we can merge.

**Comment by discdiver** [comment]
Closed as completed in another PR.

**Pull request closed without merging**